pkg/asset/machines/openstack/machines.go

```diff
@@ -37,24 +37,34 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 	if poolPlatform := pool.Platform.Name(); poolPlatform != openstack.Name {
 		return nil, fmt.Errorf("non-OpenStack machine-pool: %q", poolPlatform)
 	}
-	platform := config.Platform.OpenStack
-
-	az := ""
-	provider, err := generateProvider(clusterID, platform, pool.Platform.OpenStack, osImage, az, role, userDataSecret)
-	if err != nil {
-		return nil, err
-	}
 
-	if role == "master" {
-		provider.ServerGroupName = clusterID + "-master"
-	}
+	mpool := pool.Platform.OpenStack
+	platform := config.Platform.OpenStack
 
 	total := int64(1)
 	if pool.Replicas != nil {
 		total = *pool.Replicas
 	}
 	machines := make([]machineapi.Machine, 0, total)
 	for idx := int64(0); idx < total; idx++ {
+		zone := mpool.Zones[int(idx)%len(mpool.Zones)]
```

Comment:
What happens if len(mpool.Zones) is 0? Is it possible?

Reply:
This should not be possible. It will either be set by the user, autofilled by the installer, or set to [""] by default.

Reply:
I just patched that up. Ill add unit tests tomorrow to make sure.